src/Orleans/Serialization/ILSerializerGenerator.cs

```diff
@@ -14,12 +16,54 @@ internal class ILSerializerGenerator
         private static readonly RuntimeTypeHandle UintPtrTypeHandle = typeof(UIntPtr).TypeHandle;
 
         private static readonly TypeInfo DelegateTypeInfo = typeof(Delegate).GetTypeInfo();
+        
+        private readonly Dictionary<RuntimeTypeHandle, SimpleTypeSerializer> directSerializers;
 
         private readonly ReflectedSerializationMethodInfo methods = new ReflectedSerializationMethodInfo();
 
         private readonly SerializationManager.DeepCopier immutableTypeCopier = obj => obj;
 
         private readonly ILFieldBuilder fieldBuilder = new ILFieldBuilder();
+        
+        public ILSerializerGenerator()
+        {
+            this.directSerializers = new Dictionary<RuntimeTypeHandle, SimpleTypeSerializer>
+            {
+                [typeof(int).TypeHandle] = new SimpleTypeSerializer(w => w.Write(default(int)), r => r.ReadInt()),
+                [typeof(uint).TypeHandle] = new SimpleTypeSerializer(w => w.Write(default(uint)), r => r.ReadUInt()),
```

Comment:
aren't primitive types already covered by the built-in types in the SerializationManager? What's the reason for these in the fallback serializer?